compiler/rustc_hir_analysis/src/bounds.rs

```diff
@@ -57,7 +57,7 @@ impl<'tcx> Bounds<'tcx> {
 
     pub fn push_sized(&mut self, tcx: TyCtxt<'tcx>, ty: Ty<'tcx>, span: Span) {
         let sized_def_id = tcx.require_lang_item(LangItem::Sized, Some(span));
-        let trait_ref = ty::Binder::dummy(tcx.mk_trait_ref(sized_def_id, [ty]));
+        let trait_ref = ty::Binder::dummy(ty::TraitRef::new(tcx, sized_def_id, [ty]));
```

Comment:
All those `Binder::dummy` make me think that we may want an alias for `Binder::dummy(ty::TraitRef::new(...))`, not sure about the name tho

Reply:
a bunch of these `Binder::dummy` are actually unnecessary now because I've added a `ToPredicate` impl for `TraitRef`. Not sure whether you want to merge this as is, might be nice to remove some of these `ty::Binder::dummy` given that you're already touching this code.

Reply:
I'll try that after fixing CI.

Reply:
Okay so the 7 cases left (which matched `Binder::dummy\([ \n]*ty::TraitRef::new`) are `.without_const()`, rustdoc and some weird case in `rustc_trait_selection/src/traits/project.rs` where removing `ty::Binder` causes an error...

Reply:
Wait wait wait.

Where is this ToPredicate impl for TraitRef? I have specifically removed this before, because it hides exactly this Binder::dummy call. This point is the be *very* explicit about that, because otherwise we can easily fall back into the case of "forgot to track bound vars".

Reply:
I don't think that is true? Going to start a zulip conversation about this.